File: Solutions/34- Find First and Last Position of Element in Sorted Array/Solution.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        if (nums.length == 0)
            return new int[]{-1,-1};
        int left = 0;
        int right = nums.length - 1;
        int middle;
        int start = nums.length;
        int finish = -1;

        while (left < right) {
            middle = (left + right) / 2;
            if (target > nums[middle]) {
                left = middle + 1;
            } else if (target < nums[middle]) {
                right = middle - 1;
            } else {
                finish = middle;
                left = middle + 1;
            }
        }

        if (nums[left] == target)
            finish = left;

        left = 0;
        right = nums.length - 1;

        while (left < right) {
            middle = (left + right) / 2;
            if (target > nums[middle]) {
                left = middle + 1;
            } else if (target < nums[middle]) {
                right = middle - 1;
            } else {
                start = middle;
                right = middle - 1;
            }
        }

        if (nums[left] == target)
            start = left;

        if (start == nums.length)
            start = -1;

        return new int[]{start,finish};
    }
}

File: Solutions/34- Find First and Last Position of Element in Sorted Array/Solution.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        if not nums:
            return [-1, -1]

        left, right = 0, len(nums) - 1
        start, finish = len(nums), -1

        # Search for the finish position
        while left < right:
            middle = (left + right) // 2
            if target > nums[middle]:
                left = middle + 1
            elif target < nums[middle]:
                right = middle - 1
            else:
                finish = middle
                left = middle + 1

        if nums[left] == target:
            finish = left

        # Reset for searching the start position
        left, right = 0, len(nums) - 1

        # Search for the start position
        while left < right:
            middle = (left + right) // 2
            if target > nums[middle]:
                left = middle + 1
            elif target < nums[middle]:
                right = middle - 1
            else:
                start = middle
                right = middle - 1

        if nums[left] == target:
            start = left

        if start == len(nums):
            start = -1

        return [start, finish]